### ISSUE TITLE
Better linux shell integration.

### ISSUE THREAD

**Issue by invy** [elfmz/far2l#9]
Feature request.

Provide better linux shell integration:
 - use .profile and shell rc files to configure shell used in far
 - use $PS1 as a command prompt string
 - cd command should also change directory displayed on the current panel (cd .., cd ~, cd)

**Comment by lieff** [comment]
As I can see cd command now works.
Use of $PS1 is not good idea. Usually user already know computer and user name, omitting them save some space for useful information like long file path.
Configure shell is good feature.

**Comment by invy** [comment]
* tested cd, it's working now, eventually it got fixed.
* $PS1 it's a customization. For example I like to display current repository branch I'm in and without this information I've noticed, that I experience some discomfort. The best solution is to integrate whole shell configuration and let user decide how he wants to deal with prompt.

**Comment by elfmz** [comment]
FAR has its own prompt customization, configured via Options/Command Line Settings

**Comment by invy** [comment]
This customization is not that powerful the zsh or bash has and also it requires user to do this configuration twice: in the shell and in the far separately, which is not very convenient to say the least.

**Comment by singalen** [comment]
Items 1 and 3 already work in Far (`.bashrc` works), so they can be crossed out.

So the issue can probably be renamed to "Support $PS1"?

**Comment by elfmz** [comment]
And about PS1: far have own prompt formatting option. Just may be its not so fancy as PS1..

**Comment by singalen** [comment]
Yep, I verified it before writing. I believe that it only makes sense on Windows, and is redundant on Linux/OSX.

Windows does have `%PROMPT%`, but its features are laughable.

The way I would approach the issue is - to REMOVE Far's option (or move it to registry/whatever), and use the system default. Some minority of Windows users will not like it, but as long as they have a tool to customize it, this should be OK.

**Comment by ujos** [comment]
>  For example I like to display current repository branch I'm in and without this information I've noticed, that I experience some discomfort. 

If this issue is still actual, try to build `far2l` from this branch `https://github.com/ujos/far2l/tree/get_git_branch`.

**Comment by hadmir** [comment]
It would be nice to have option to color git branch, so it is not in the same color as path. Like $z[ff0000] for red.

**Comment by ujos** [comment]
> It would be nice to have option to color git branch, so it is not in the same color as path. Like $z[ff0000] for red.

Well, I think it would be better to add this functionality to the command line prompt parser rather than to the Git branch function, so you can modify colors not just for the Git branch itself

**Comment by hadmir** [comment]
FAR on windows is displaying Git branch in different color then the rest of the prompt, but yes, setting color for each part would be awesome.
